src/program/downloaders/alldebrid.py

```diff
@@ -1,48 +1,31 @@
-import contextlib
-import time
 from datetime import datetime
-from os.path import splitext
-from pathlib import Path
-from types import SimpleNamespace
-from typing import Generator, List
 
+from loguru import logger
+from program.settings.manager import settings_manager as settings
 from requests import ConnectTimeout
-from RTN.exceptions import GarbageTorrent
-from RTN.extras import extract_episodes
-from RTN.parser import parse
-from sqlalchemy.orm import lazyload
-
-from program.db.db import db
-from program.db.db_functions import get_stream_count, load_streams_in_pages
-from program.media.item import Episode, MediaItem, Movie, Season, Show
-from program.media.state import States
-from program.media.stream import Stream
-from program.settings.manager import settings_manager
-from utils.logger import logger
+from utils import request
 from utils.ratelimiter import RateLimiter
-from utils.request import get, ping, post
 
-WANTED_FORMATS = {".mkv", ".mp4", ".avi"}
+from .shared import VIDEO_EXTENSIONS, FileFinder, premium_days_left
+
 AD_BASE_URL = "https://api.alldebrid.com/v4"
 AD_AGENT = "Riven"
-AD_PARAM_AGENT = f"agent={AD_AGENT}"
+
+inner_rate_limit = RateLimiter(12, 1)  # 12 requests per second
+overall_rate_limiter = RateLimiter(600, 60)  # 600 requests per minute
+
 
 class AllDebridDownloader:
     """All-Debrid API Wrapper"""
 
     def __init__(self):
-        self.rate_limiter = None
         self.key = "alldebrid"
-        self.settings = settings_manager.settings.downloaders.all_debrid
-        self.download_settings = settings_manager.settings.downloaders
-        self.auth_headers = {"Authorization": f"Bearer {self.settings.api_key}"}
-        self.proxy = self.settings.proxy_url if self.settings.proxy_enabled else None
-        self.inner_rate_limit = RateLimiter(12, 1)  # 12 requests per second
-        self.overall_rate_limiter = RateLimiter(600, 60)  # 600 requests per minute
+        self.settings = settings.settings.downloaders.all_debrid
         self.initialized = self.validate()
-        if not self.initialized:
-            return
-        logger.success("AllDebrid initialized!")
+        if self.initialized:
+            self.existing_hashes = [torrent["hash"] for torrent in get_torrents()]
+            self.file_finder = FileFinder("filename", "filesize")
```

Comment:
AllDebrid uses "n" and "s", but other parts of the code base still assume "filename" so that configurability can't be used yet